src/molecule_plugins/openstack/playbooks/create.yml

```diff
@@ -0,0 +1,170 @@
+      when:
+        - item.0.security_group is defined
+        - item.0.security_group.create | default(true)
+        - "'rules' in item.0.security_group"
+      loop: "{{ molecule_yml.platforms | subelements('security_group.rules', skip_missing=True) }}"
+
+    - name: Create network
+      openstack.cloud.network:
+        name: "molecule-test-{{ item.network.name }}-{{ uuid }}"
+        state: present
+        external: "{{ item.network.external | default(omit) }}"
+      when:
+        - item.network is defined
+        - item.network.name is defined
+        - item.network.create | default(true)
+      loop: "{{ molecule_yml.platforms }}"
+
+    - name: Create subnet in network
+      openstack.cloud.subnet:
+        name: "molecule-test-{{ item.network.subnet.name }}-{{ uuid }}"
+        state: present
+        network_name: "molecule-test-{{ item.network.name }}-{{ uuid }}"
+        cidr: "{{ item.network.subnet.cidr }}"
+        ip_version: "{{ iitem.network.subnet.ipv | default(4) | int }}"
+        dns_nameservers: "{{ item.network.subnet.dns_nameservers | default(omit) }}"
+        host_routes: "{{ item.network.subnet.host_routes | default(omit) }}"
+      when:
+        - item.network is defined
+        - item.network.name is defined
+        - item.network.create | default(true)
+        - item.network.subnet is defined
+        - item.network.subnet.name is defined
+      loop: "{{ molecule_yml.platforms }}"
+
+    - name: Create Router
+      openstack.cloud.router:
+        name: "molecule-test-{{ item.network.router.name }}-{{ uuid }}"
+        state: present
+        network: "{{ item.network.router.ext_network | default(omit) }}"
+        enable_snat: "{{ item.network.router.snat | default(omit) }}"
+        interfaces:
+          - net: "molecule-test-{{ item.network.name }}-{{ uuid }}"
+            subnet: "molecule-test-{{ item.network.subnet.name }}-{{ uuid }}"
+      when:
+        - item.network is defined
+        - item.network.name is defined
+        - item.network.create | default(true)
+        - item.network.router is defined
+        - item.network.router.name is defined
+        - item.network.subnet is defined
+        - item.network.subnet.name is defined
+      loop: "{{ molecule_yml.platforms }}"
+
+    - name: Create ssh key
+      openstack.cloud.keypair:
+        name: "{{ key_name }}"
+        state: present
+      register: key_pair
+
+    - name: Persist identity file
+      ansible.builtin.copy:
+        dest: "{{ identity_file }}"
+        content: "{{ key_pair.keypair.private_key }}"
+        mode: "0600"
+      when: key_pair is changed # noqa no-handler
+
+    - name: Create openstack instance
+      openstack.cloud.server:
+        state: present
+        name: "molecule-test-{{ item.name }}-{{ uuid }}"
+        description: "{{ item.description | default('Molecule test instance') }}"
+        image: "{{ item.image }}"
+        key_name: "{{ key_name }}"
+        flavor: "{{ item.flavor }}"
```

Comment:
adding volume support which is needed if no-disk flavors are used would be good, imo

```
diff --git a/src/molecule_plugins/openstack/playbooks/create.yml b/src/molecule_plugins/openstack/playbooks/create.yml
index 7d01a7d..f71f5c8 100644
--- a/src/molecule_plugins/openstack/playbooks/create.yml
+++ b/src/molecule_plugins/openstack/playbooks/create.yml
@@ -102,6 +102,10 @@
         image: "{{ item.image }}"
         key_name: "{{ key_name }}"
         flavor: "{{ item.flavor }}"
+        volume_size: "{{ item.volume_size | default(omit) }}"
+        boot_from_volume: "{{ item.boot_from_volume | default(false) }}"
+        terminate_volume: "{{ item.terminate_volume | default(false) }}"
+        auto_ip: "{{ item.auto_ip | default(true) }}"
         network: >-
           {{
             'molecule-test-' + item.network.name + '-' + uuid
```

Reply:
Yes, you are right. I will add this in the next few days